recipes/gmp/all/conanfile.py

```diff
@@ -49,19 +67,18 @@ def package_id(self):
 
     def build_requirements(self):
         self.build_requires("m4/1.4.18")
-        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/20200517")
+        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
         if self.settings.compiler == "Visual Studio":
             self.build_requires("yasm/1.3.0")
-            self.build_requires("automake/1.16.3")
+            self.build_requires("automake/1.16.4")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
 
+    @functools.lru_cache(1)
```

Comment:
Is this needed everywhere?

Reply:
Not needed, only if you want to be fancy. 😅✌️